eitprocessing/datahandling/datacollection.py

```diff
@@ -111,3 +112,24 @@ def concatenate(self: Self[V], other: Self[V]) -> Self[V]:
             concatenated[key] = self[key].concatenate(other[key])
 
         return concatenated
+
+    def select_by_time(
+        self,
+        start_time: float | None,
+        end_time: float | None,
+        start_inclusive: bool = True,
+        end_inclusive: bool = False,
+    ) -> Self:
+        """Return a DataCollection containing sliced copies of the items."""
+        return DataCollection(
+            self.data_type,
+            **{k: v.select_by_time(start_time, end_time, start_inclusive, end_inclusive) for k, v in self.items()},
+        )
+
+    @property
+    def t(self):
+        """Time indexer.
+
+        See slicing.TimeIndexer.
+        """
+        return TimeIndexer(self)
```

Comment:
This would make most sense, except that 1. `SelectByTime` has the `time` attribute and 2. `SelectByTime` inherits from `SelectByIndex`, while selecting by index is not possible for the `DataCollection`. 

**Time attribute** 
We probably could remove that, but it makes sense to me to have the explicit `time` attribute when `select_by_time` depends on it.

On the other hand, `select_by_time` explicitly checks for the existence of the time attribute. So this this would work all the same.

**SelectByIndex**
This is not easily fixable. An alternative would be that we create another class names `HasTimeIndexer` (the name `TimeIndexer` is already taken by de indexer itself) that only contains the method `t()`. both `SelectByTime` and `DataCollection` can then inherit from `HasTimeIndexer`. 

I added these changes to d893c377c96b11022152aa3cad43b4976c66caa9. We can revert it if this makes it too complex.

Reply:
why not add the property to the `TimeIndexer` class itself and inherit from there?

Also, see [this comment](https://github.com/EIT-ALIVE/eitprocessing/pull/168#issuecomment-2042227897) re: slicing a sequence by time:

> Indeed, without a dedicated time axis to `Sequence` it's odd to slice by index. That being said, it is a very useful operation that users will be used to using for other similar data. Hadn't we discussed at some point to tag one data source as `primary_data` or something, defaulting to `EITData` if present? If so, that could be the axis for slice by index. We can also allow an optional argument to the select by index method, indicating which data's time axis to use for it.

Similarly, we could give `DataCollection` an optional primary data source, defaulted to `None`. To slice a `DataCollection` by index, it needs to have a one defined, otherwise it does raises an error.